site/src/components/pages/user/UserPageComponent.tsx

```diff
@@ -27,44 +28,39 @@ import { Button } from "../../Button";
 import { ListViewCard } from "./ListViewCard";
 import { OverviewCard } from "./OverviewCard";
 import { Sidebar } from "./Sidebar";
-import { TABS, TabHeader, TabPanel } from "./Tabs";
+import { TABS, TabHeader, TabPanel, TabValue } from "./Tabs";
 import { Modal } from "../../Modal";
 import { TextField } from "../../TextField";
 
 const SIDEBAR_WIDTH = 300;
 
 export type UserPageProps = {
-  user: SerializedUser;
   blocks: ExpandedBlockMetadata[];
   entityTypes: EntityType[];
+  initialActiveTab: TabValue;
+  user: SerializedUser;
 };
 
 export const UserPageComponent: VoidFunctionComponent<UserPageProps> = ({
   blocks,
   entityTypes,
+  initialActiveTab,
   user,
 }) => {
   const router = useRouter();
 
-  const activeTab = useMemo(() => {
+  useEffect(() => {
     const { profileTabs } = router.query;
 
-    if (!profileTabs?.length) {
-      return TABS[0].value;
-    }
-
     const matchingTab = TABS.find((tab) => tab.slug === profileTabs?.[0]);
 
-    if (!matchingTab) {
-      if (process.browser) {
-        void router.replace(`/@${user.shortname}`);
-      }
-      return TABS[0].value;
+    if (!matchingTab && typeof window !== "undefined") {
```

Comment:
`typeof window !== "undefined"` should be unnecessary as `useEffect` does not run on the server.

Reply:
Should be done now :D